Split build & deploy GitHub Actions, Edit tutorials

billy1624 opened the pull request:
## Changes

- Split build & deploy GitHub Actions
  - Build: build mdbook of all tutorials (runs on PR and master branch)
  - Deploy: build and deploy all mdbook as GitHub Pages (only runs on master branch)
- Editing SeaORM tutorials (in progress)
- Auto publish all mdbook when new commit pushed to master
  - SimpleCRUD can be access on
    - [/sea-orm-tutorial](https://billy1624.github.io/sea-orm-tutorial/) (will be serving the homepage once it's ready)
    - [/sea-orm-tutorial/SimpleCRUD](https://billy1624.github.io/sea-orm-tutorial/SimpleCRUD)
  - SeaORM-TODO-App can be access on
    - [/sea-orm-tutorial/SeaORM-TODO-App](https://billy1624.github.io/sea-orm-tutorial/SeaORM-TODO-App)

## References

- Generating mdbook sitemap
  - rust-lang/mdBook#1491

billy1624 committed:
LICENSE

billy1624 committed:
Add SimpleCRUD/.gitignore

billy1624 committed:
Add SimpleCRUD/book/.gitignore

billy1624 committed:
Remove ignored files

billy1624 committed:
Rename `SeaORM-TODO-App/SeaORM-TODO-Book` to `SeaORM-TODO-App/book`

billy1624 committed:
Open a separate workspace for each tutorial

billy1624 committed:
Edit book.toml

billy1624 committed:
Split GitHub actions into build & deploy

billy1624 committed:
Edit

billy1624 committed:
Edit

charleschege commented on LICENSE:
```diff
@@ -0,0 +1,201 @@
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
@billy1624 This part needs some change. The Name of the Organization and year of copyright

billy1624 replied:
Thanks!! @charleschege

billy1624 commented on SeaORM-TODO-App/book/src/tcp-server.md:
```diff
@@ -18,23 +18,19 @@
-   
-
-2. Add `anyhow` crate for error handling
+1. Add `anyhow` crate for error handling
```
Hey @charleschege, I was thinking to group steps into subheadings. Such as...

1. Setting up file structure for TCP Server
1. Installing necessary dependencies
1. Creating database in PostgreSQL
1. Configuring the environment variables

charleschege committed:
Add details to the Apache-2.0 license

charleschege reviewed (commented):
LGTM. Will make the subheadings

charleschege committed:
Move to subheadings

Change this section to use sub headings to offer a description of it's contents

charleschege reviewed (commented):
@billy1624 I have made a few changes for the LICENSE file and added sub-headings

billy1624 commented:
Hey @charleschege, I will be merging this PR now. As I already updated the sitemap of the root domain, https://www.sea-ql.org/sitemap.xml, to include https://www.sea-ql.org/sea-orm-tutorial/sitemap.xml.

billy1624 commented:
Then, I will open another PR to continue editing the tutorials
